parsing/lexer.mll

```diff
@@ -108,6 +108,25 @@ let store_string_char c = Buffer.add_char string_buffer c
 let store_string_utf_8_uchar u = Buffer.add_utf_8_uchar string_buffer u
 let store_string s = Buffer.add_string string_buffer s
 let store_lexeme lexbuf = store_string (Lexing.lexeme lexbuf)
+let store_normalized_newline () =
+  (* #12502: we normalize newlines to "\n" at lexing time,
+     to avoid behavior difference due to OS-specific
+     newline characters in string literals.
+
+     (Example of weird behavior: Git for Windows will helpful
```

Comment:
```suggestion
     (Example: Git for Windows by default will 
```
I'm not sure the slightly derogatory "weird behavior" or "helpful" ("helpfully") are adding to the explanation.